event-service/src/error/types/badRequest.error.ts

```diff
@@ -0,0 +1,24 @@
+import { BaseError } from "../base.error";
+
+export default class BadRequestError extends BaseError {
```

Comment:
I thought about your implementation of errors in previous homework but now I am certain that you should move the logic of specific error classes to base class to avoid code duplication. Error classes BadRequest and AlreadyExist are identical except the super call with message, so you can add only super call to specific error classes and this will be enough. Also I don't understand the purpose of setPrototypeOf method, if you try to set prototype to pass instanceof check, then it is useless and you can just remove it without any affects